### PR TITLE
Fixes issue #4072 (correct IPv6 regex)

### PR DESCRIPTION
**Pull request by ylmrx**
#### Description

Fixes issue #4072 : Replace the IP regex so it covers all hex range (0-9 -> 0-f)

#### Checklist

 - [x] I have updated tests where applicable.
 - [x] I have added an entry to the CHANGELOG.

### CODE REVIEW

File: mitmproxy/tools/web/app.py [hunk 1]
```diff
@@ -499,7 +499,7 @@ def __init__(self, master: "mitmproxy.tools.web.master.WebMaster", debug: bool)
         self.add_handlers("dns-rebind-protection", [(r"/.*", DnsRebind)])
         self.add_handlers(
             # make mitmweb accessible by IP only to prevent DNS rebinding.
-            r'^(localhost|[0-9.:\[\]]+)$',
+            r'^(localhost|[\da-f.:\[\]]+)$',
```
Comment (Kriechi) [inline mitmproxy/tools/web/app.py:502]: I think this would allow `abcdef.ac` which is a valid domain + TLD - and therefore not really what we want here?

Comment (ylmrx) [reply mitmproxy/tools/web/app.py:502]: It's incomplete (it validated "999...1234.123" as an IP for example) as it was before.
A full RE (OP provided one in #4072), which reliably validates all IPv4 and 6, will be really complicated 
(imho, it shouldn't rely on a regex for this purpose).

Comment (tunnelpr0) [reply mitmproxy/tools/web/app.py:502]: I also think that this is quite a bullshit and you are transforming a "normal" issue to a security issue...

Please use the longer regex pattern - i am pretty sure that it wont cause issues!

Comment (ylmrx) [reply mitmproxy/tools/web/app.py:502]: I get this is the first time you submit an issue. You should know this language is way outside the line.
Feel free to review and put the big RE. 
I'm out.

**Comment by ylmrx** [comment]
duplicate. follow-up in #4074

**Comment by mhils** [comment]
Thanks @ylmrx! 

@tunnelpr0, we all try to be friendly here. Please turn down your language.

**Pull request closed without merging**